src/rgw/rgw_sal_store.h

```diff
@@ -245,6 +246,8 @@ class StoreObject : public Object {
        * work with lifecycle */
       return -1;
     }
+    virtual jspan_context& get_trace() { return trace_ctx; }
+    virtual void set_trace (jspan_context&& _trace_ctx) { trace_ctx = _trace_ctx; }
```

Comment:
```suggestion
    virtual void set_trace (jspan_context&& _trace_ctx) { trace_ctx = std::move(_trace_ctx); }
```

Without the `std::move` the `_trace_ctx` becomes a l-value reference, so `operator=(T&)` is going to be called.